azure/services/managedclusters/spec.go

```diff
@@ -0,0 +1,430 @@
+	APIServerAccessProfile *APIServerAccessProfile
+
+	// Headers is the list of headers to add to the HTTP requests to update this resource.
+	Headers map[string]string
+}
+
+// AADProfile is Azure Active Directory configuration to integrate with AKS, for aad authentication.
+type AADProfile struct {
+	// Managed defines whether to enable managed AAD.
+	Managed bool
+
+	// EnableAzureRBAC defines whether to enable Azure RBAC for Kubernetes authorization.
+	EnableAzureRBAC bool
+
+	// AdminGroupObjectIDs are the AAD group object IDs that will have admin role of the cluster.
+	AdminGroupObjectIDs []string
+}
+
+// AddonProfile is the profile of a managed cluster add-on.
+type AddonProfile struct {
+	Name    string
+	Config  map[string]string
+	Enabled bool
+}
+
+// SKU is an AKS SKU.
+type SKU struct {
+	// Tier is the tier of a managed cluster SKU.
+	Tier string
+}
+
+// LoadBalancerProfile is the profile of the cluster load balancer.
+type LoadBalancerProfile struct {
+	// Load balancer profile must specify at most one of ManagedOutboundIPs, OutboundIPPrefixes and OutboundIPs.
+	// By default the AKS cluster automatically creates a public IP in the AKS-managed infrastructure resource group and assigns it to the load balancer outbound pool.
+	// Alternatively, you can assign your own custom public IP or public IP prefix at cluster creation time.
+	// See https://docs.microsoft.com/en-us/azure/aks/load-balancer-standard#provide-your-own-outbound-public-ips-or-prefixes
+
+	// ManagedOutboundIPs are the desired managed outbound IPs for the cluster load balancer.
+	ManagedOutboundIPs *int32
+
+	// OutboundIPPrefixes are the desired outbound IP Prefix resources for the cluster load balancer.
+	OutboundIPPrefixes []string
+
+	// OutboundIPs are the desired outbound IP resources for the cluster load balancer.
+	OutboundIPs []string
+
+	// AllocatedOutboundPorts are the desired number of allocated SNAT ports per VM. Allowed values must be in the range of 0 to 64000 (inclusive). The default value is 0 which results in Azure dynamically allocating ports.
+	AllocatedOutboundPorts *int32
+
+	// IdleTimeoutInMinutes  are the desired outbound flow idle timeout in minutes. Allowed values must be in the range of 4 to 120 (inclusive). The default value is 30 minutes.
+	IdleTimeoutInMinutes *int32
+}
+
+// APIServerAccessProfile is the access profile for AKS API server.
+type APIServerAccessProfile struct {
+	// AuthorizedIPRanges are the authorized IP Ranges to kubernetes API server.
+	AuthorizedIPRanges []string
+	// EnablePrivateCluster defines hether to create the cluster as a private cluster or not.
+	EnablePrivateCluster *bool
+	// PrivateDNSZone is the private dns zone for private clusters.
+	PrivateDNSZone *string
+	// EnablePrivateClusterPublicFQDN defines whether to create additional public FQDN for private cluster or not.
+	EnablePrivateClusterPublicFQDN *bool
+}
+
+var _ azure.ResourceSpecGetterWithHeaders = (*ManagedClusterSpec)(nil)
```

Comment:
What's this line for?

Reply:
it's validating that the managed cluster spec implements the `ResourceSpecGetterWithHeaders` interface

it's a common go pattern to check that a type implements an interface https://golangbyexample.com/go-check-if-type-implements-interface/